r-package/grf/R/input_utilities.R

```diff
@@ -11,6 +11,55 @@ validate_X <- function(X) {
   if (inherits(X, "Matrix") & !(inherits(X, "dgCMatrix"))) {
       stop("Currently only sparse data of class 'dgCMatrix' is supported.")
   }
+  
+  if (any(is.na(X))){
+    stop("The feature matrix X contains at least one NA.")
+  }
+}
+
+validate_YX <- function(Y,X) {
+  validate_Y(Y)
+  if(length(Y) != nrow(X)) { stop("length(Y) does not equal nrow(X).") }
+}
+
+validate_Y <- function(Y) {
+  if(!is.vector(Y) || !is.numeric(Y)) {
+    stop(paste("The outcome Y must be numeric vector. GRF does not", 
+               "currently support non-numeric or non-vector outcomes."))
+  }
+
+  if (any(is.na(Y))){
+    stop("The outcome vector Y contains at least one NA.")
+  }
+}
+
+validate_WX <- function(W,X) {
+  validate_W(W)
+  if(length(W) != nrow(X)) { stop("length(W) does not equal nrow(X).") }
+}
+
+validate_W <- function(W) {
+  if(!is.vector(W) || !is.numeric(W)) {
+    stop(paste("The treatment W must be numeric vector. GRF does not", 
+               "currently support non-numeric or non-vector treatments."))
+  }
+
+  if(length(unique(W))!=2) {
```

Comment:
I think we should omit this check, as we actually allow for continuous treatments.

Since `validate_Y`, `validate_W` will contain the same logic after removing the check, I think it'd be nice to have a single function `validate_observations(...)` that takes all the observation vectors and validates each one. It would also be nice to simplify and just have one version of method instead of two (right now we have `validate_WX` and `validate_W` for example, when just one method would suffice).